prow/README.md

```diff
@@ -1,67 +1,90 @@
+
+<!-- Update the folder structure each time you modify it. -->
+
+Its structure looks as follows:
 
-```
-gcloud container clusters get-credentials {CLUSTER_NAME} --zone={ZONE_NAME} --project={PROJECT_NAME}
 ```
 
-2. Export the environment variables:
- - **BUCKET_NAME** is a GCS bucket in the Google Cloud project that is used to store Prow Secrets.
- - **KEYRING_NAME** is the KMS key ring.
- - **ENCRYPTION_KEY_NAME** is the key name in the key ring that is used for data encryption.
-
- ```
- export BUCKET_NAME=kyma-prow
- export KEYRING_NAME=kyma-prow
- export ENCRYPTION_KEY_NAME=kyma-prow-encryption
- ```
- 
-3. Run the following script to start the installation process:
-
-```bash
-./install-prow.sh
+  ├── cluster               # All "yaml" files for Prow cluster provisioning.           
+  ├── images                # Images for component jobs that you can also use for generic builds.                                             
+  ├── jobs                  # All files with jobs definitions.
+  ├── scripts               # Scripts used by the test jobs.
+  ├── config.yaml           # The main Prow configuration, without job definitions. For example, it contains Plank configuration and Preset definitions.
+  ├── install-prow.sh       # The script for the production cluster installation.
+  └── plugins.yaml          # The file with Prow plugins configuration.
 ```
 
-The installation script performs the following steps to install Prow:
+## Installation
+
+Read the [`docs`](../docs/prow/README.md) to lean how to configure the production Prow or install it on your forked repository for development and testing.
+
+## Development
+
+Read about the conventions for organizing and naming jobs in the `prow` subdirectories.
+
+### Strategy for organising jobs
+
+The `jobs/{repository_name}` directories have subdirectories which represent each component and contain job definitions. Each file must have a unique name. Job definitions not connected to a particular component, like integration jobs, are defined directly under the `jobs/{repository_name}` directory.
+
+For example:
+
+   ```
+   ...
+   prow
+   |- cluster
+   | |- starter.yaml
+   |- images
+   |- jobs
+   | |- kyma
+   | | |- components
+   | | | |- environments
+   | | | | |- environments.jobs.yaml
+   | | |- kyma.integration.yaml
+   |- scripts
+   |- config.yaml
+   |- plugins.yaml
+   ...
+   ```
+
+### Convention for naming jobs
 
-- Deploy the NGINX Ingress Controller.
-- Create a ClusterRoleBinding.
-- Deploy Prow components with the `a202e595a33ac92ab503f913f2d710efabd3de21`revision.
-- Deploy Cert Manager.
-- Deploy Secure Ingress.
-- Removes Insecure Ingress.
+When you define jobs for Prow, both **name** and **context** of the job must follow one of these patterns:
 
-To check if the installation is successful, perform the following steps:
+- `prow/{repository_name}/{component_name}/{job_name}` for components
```

Comment:
job name should not be a prefix of another job. We had such situation for kyma-integration and kyma-integration-gke jobs: when you wrote /retest kyma-integration both of them were executed. 
Also we should define name for the release presubmit job, in my PR I proposed the following approach:
`prow/release/kyma/components/ui-api-layer`
And what about job name for postsubmit jobs? @michal-hudy

Reply:
I'm not sure if we need that for `postsubmit` jobs. In Prow you can see it was a job executed on `master`. For `release` it wasn't that easy to recognize, because it is `presubmit` job so it looks in Prow as the job for PR.